content/en/docs/armory-agent/armory-agent-quick.md

```diff
@@ -5,48 +5,146 @@ description: >
-* If Agent is running in any of the other modes, then the `ClusterRole` or `Role` is the one the `kubeconfigFile` uses to interact with the target cluster. `kubeconfigFile` is configured in `kubesvc.yml` of the Agent pod.
+This step is performed in the cluster Spinnaker service is running. You will add the Clouddriver plugin and expose it as type `LoadBalancer` on gRPC port `9091`. In step 2, the Agent will be configured to communicate with Clouddriver. _Take note to ensure the plugin version is compatible with the Spinnaker version. See the comments in the manifest_.
 
-Example configuration for deploying `Pod` manifests:
+Add this manifest to your Kustomize patches:
 
-{{< tabs name="agent-permissions" >}}
-{{% tab name="ClusterRole" %}}
+```yaml
+# The plugin version (see kubesvc-plugin below) must be comatible with the spinnaker version, check here: https://docs.armory.io/docs/armory-agent/armory-agent-quick/#compatibility-matrix
+# Change "spinnaker" name below if your spinsvc is called something else
+apiVersion: spinnaker.armory.io/v1alpha2
+kind: SpinnakerService
+metadata:
+  name: spinnaker
+spec:
+  spinnakerConfig:
+    profiles:
+      clouddriver:
+        spinnaker:
+          extensibility:
+            pluginsRootPath: /opt/clouddriver/lib/plugins
+            plugins:
+              Armory.Kubesvc:
+                enabled: true
+                extensions:
+                  armory.kubesvc:
+                    enabled: true
+        # Plugin config
+        kubesvc:
+          cluster: redis
+#          eventsCleanupFrequencySeconds: 7200
+#          localShortCircuit: false
+#          runtime:
+#            defaults:
+#              onlySpinnakerManaged: true
+#            accounts:
+#              account1:
+#                customResources:
+#                  - kubernetesKind: MyKind.mygroup.acme
+#                    versioned: true
+#                    deployPriority: "400"
+  kustomize:
+    clouddriver:
+      deployment:
+        patchesStrategicMerge:
+          - |
+            spec:
+              template:
+                spec:
+                  initContainers:
+                  - name: kubesvc-plugin
+                    image: docker.io/armory/kubesvc-plugin:0.8.9 # <-- Version number must be compatible with spinnaker version
+                    volumeMounts:
+                      - mountPath: /opt/plugin/target
+                        name: kubesvc-plugin-vol
+                  containers:
+                  - name: clouddriver
+                    volumeMounts:
+                      - mountPath: /opt/clouddriver/lib/plugins
+                        name: kubesvc-plugin-vol
+                  volumes:
+                  - name: kubesvc-plugin-vol
+                    emptyDir: {}
+
+```
+
+To expose Clouddriver for remote Agents, add this manifest to your Kustomize resources:
+
+```yaml
+# This loadbalancer service exposes the gRPC port on cloud-driver for the remote agents to connect
```

Comment:
This is not just about remote agents. All agents require a service defined to be able to reach to clouddriver. You can then direct an ingress to the service for remote agents.

Reply:
when agent is running on same cluster as clouddriver, it can just talk via clusterIP. this is technically a expose method within the cluster. I will think about how to reword this. For simplicity, I'm using the K8s LB config. I can add a note that various cloud providers may require aditional annotation to work.